hazelcast/src/main/java/com/hazelcast/client/impl/protocol/ClientMessage.java

```diff
@@ -328,7 +328,9 @@ public boolean readFrom(ByteBuffer src) {
                 // we don't have even the frame length ready
                 return false;
             }
-            frameLength = Bits.readIntL(src.array(), src.position());
+            frameLength = Bits.readIntL(src);
```

Comment:
what about directly reading from byteBuffer and get rid of `Bits.readIntL(ByteBuffer)`

```
            src.order(ByteOrder.LITTLE_ENDIAN);
            frameLength = src.getInt(src.position());
```